### PR TITLE
Feature/small fixes nino

### PR DESCRIPTION
**Pull request by NinoRy**
#

### CODE REVIEW

**Comment by NinoRy** [comment]
@schmto @cschra @xee8ai 

Note: This comment refers to the comments from branch https://github.com/nmsprime/nmsprime/pull/622. I just move the commits here because of the reason explained in last comment there.

To the topic: 
What else should the user think when he sees 3 modems in the panel 'Modems' on the contract page? That they are related to to another contract? I think it's just superfluous and nobody would expect modems from another contract there. But that's just my opinion.

**Comment by cschra** [comment]
I just clicked through the demo system to see if it really mattered and found the "Assigned/Zugewiesene" not necessary. It is indeed self expainatory.

File: resources/views/bootstrap/panel.blade.php [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
+
+	$display = isset($options['display']) ? 'display: '.$options['display'] : '';
 ?>
 
 {{-- begin col-dyn --}}
 <div class="col-md-{{$md}} ui-sortable">
 	<div class="panel panel-inverse card-2 d-flex flex-column" data-sortable-id="table-index-{{ isset($i) ? $i : '1'}}">
 		@include ('bootstrap.panel-header', ['view_header' => $view_header])
-		<div class="panel-body fader d-flex flex-column" style="overflow-y:{{ $overflow_y }}; {{ $style }}">
+		<div class="panel-body fader d-flex flex-column" style="overflow-y:{{ $overflow_y }}; {{ $style }}; {{ $display }}">
```
Comment (cschra) [inline resources/views/bootstrap/panel.blade.php]: If you touch this part of the code, i would recommend unifying it.
Also as a personal preference I would inline the php stuff
```php
    <div class="panel-body fader d-flex flex-column" style="{{ $overflowY }}; {{ $style }}; {{ $display }}">
// or 
    <div class="panel-body fader d-flex flex-column" 
             style="overflow-y:{{ $overflowY }}; height: {{ $style }};display: {{ $display }}">
// or (my prefered option (with removal of the php part above) - as it is only used once
    <div class="panel-body fader d-flex flex-column" 
             style="overflow-y: {{ $overflow ?? 'auto' }}; 
                     height: {{ isset($height) ? (is_numeric($height) ? $height.'%' : 'auto') : '100%' }};  // not pretty, but it does the trick
                     display: {{ $options['display'] ?? 'inherit' }};"> // unset or initial would also be an option
```

File: modules/ProvMon/Resources/views/analyses.blade.php [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 @section('content_dash')
+	<div class="btn pull-right">
+		@include('Generic.documentation', ['documentation' => $modem->help])
+	</div>
+
 	@if ($dash)
-		@foreach ($dash as $info)
-			<div class="note note-{{$info['bsclass']}} fade in m-b-15"> {{ $info['text'] }} </div>
+		@foreach ($dash as $key => $info)
+			<div class="alert alert-{{$info['bsclass']}} fade show" style="width: 83%"> {{ $info['text'] }} </div>
```
Comment (cschra) [inline modules/ProvMon/Resources/views/analyses.blade.php]: where do the 83% come from? why not use a css class like "col-md-9"?

File: modules/ProvMon/Http/Controllers/ProvMonController.php [hunk 3]
```diff
@@ -131,11 +131,19 @@ public function analyses($id)
 
         // Log dhcp (discover, ...), tftp (configfile or firmware)
         $search = $ip ? "$mac|$modem->hostname|$ip " : "$mac|$modem->hostname";
+
+        // NOTE: This function takes a long time if syslog file is large - 0.4 to 0.6 sec
```
Comment (cschra) [inline modules/ProvMon/Http/Controllers/ProvMonController.php]: use caching for this?

File: modules/ProvBase/Entities/Contract.php [hunk 4]
```diff
@@ -119,14 +120,23 @@ public function view_has_many()
...
 
         if (\Module::collections()->has('BillingBase')) {
+            // view has many version 2
             $ret['Billing']['Item']['class'] = 'Item';
             $ret['Billing']['Item']['relation'] = $this->items;
             $ret['Billing']['SepaMandate']['class'] = 'SepaMandate';
             $ret['Billing']['SepaMandate']['relation'] = $this->sepamandates;
+
+            // Show invoices in 2 panels - 2nd panel with old invoices collapsed and in 2 columns
+            $p1 = $this->invoices()->orderBy('id', 'desc')->take(15)->get();
```
Comment (cschra) [inline modules/ProvBase/Entities/Contract.php]: why 15? wouldn't it be good to have a protected variable/ config / environment value set these limits?

File: resources/views/bootstrap/panel.blade.php (hunk 1 above)
Comment (NinoRy) [reply resources/views/bootstrap/panel.blade.php]: i dont want to rewrite the whole code. I just wanted to add the display attribute. Rewriting the whole code is another task and would make unclear what happened in the commit. I would have to make another commit at least. We could definitely better structure this blade like the most. But then we should do this in a consistent way with all the blades. That`s more like a rewrite of the existing blade structure what we of course should do somewhen

File: modules/ProvMon/Resources/views/analyses.blade.php (hunk 2 above)
Comment (NinoRy) [reply modules/ProvMon/Resources/views/analyses.blade.php]: (y) didn't know it

File: modules/ProvMon/Http/Controllers/ProvMonController.php (hunk 3 above)
Comment (NinoRy) [reply modules/ProvMon/Http/Controllers/ProvMonController.php]: caching doesnt really make sense here ... it's just a comment in a first step to be able to have a quick overview what takes a lot time ... for future performance works

File: modules/ProvBase/Entities/Contract.php (hunk 4 above)
Comment (NinoRy) [reply modules/ProvBase/Entities/Contract.php]: we could discuss this ... this is just quick and dirty ;) ... let us wait for feedback of the accounting staff

File: modules/BillingBase/Entities/SepaMandate.php [hunk 5]
```diff
@@ -49,18 +49,31 @@ public function view_index_label()
...
+     * Get bootstrap class for colorization of model in GUI
+     *
+     * @return string
+     */
     public function get_bsclass()
     {
-        $bsclass = 'success';
+        // Dont check when table header of index page is assembled
+        if (! $this->id) {
+            return;
```
[the remark this answers is not in the document]
Comment (NinoRy) [reply modules/BillingBase/Entities/SepaMandate.php]: done

**Pull request merged**